src/firefly/java/edu/caltech/ipac/table/PrimitiveList.java

```diff
@@ -20,17 +21,35 @@
     int size();
     void clear();
     void trimToSize();
+    List<Class> numberTypeList = Arrays.asList(new Class[] {Byte.class, Short.class, Integer.class});
 
     default void add(Object val) {
         set(size(), val);
     }
 
     default void checkType(Object val) {
-        if (val != null && !val.getClass().isAssignableFrom(getDataClass())) {
+        int vIdx = val == null ? -1 : numberTypeList.indexOf(val.getClass());
+        int dIdx = numberTypeList.indexOf(getDataClass());
+
+        // number type value or non number type value
+        if ((vIdx >= 0 && dIdx >= 0 && vIdx > dIdx) ||
+            (vIdx < 0 && val != null && !val.getClass().isAssignableFrom(getDataClass()))) {
```

Comment:
We've talked about this, and I was unsure of the solution.  After looking at it now, I think the solution is to check at the point where the casting happens.  checkType() is not working nor is it needed.